wordpress-mobile/WordPress-Android
# [Site Design Revamp] Re-enable large recommended theme screenshots

Fixes #16847

## Description

This PR re-enables the larger screenshots disabled as a work-around in https://github.com/wordpress-mobile/WordPress-Android/pull/16747. Formerly, the `scale` parameter was not being passed (or defaulted to 1).

Since this parameter is needed to achieve properly sized screenshots for higher screen densities, this PR passes this value based on the device's screen density. Since the dimensions are already scaled by this value (by default when retrieving the dimension resource in Android), this PR manually. But, since the backend _also_ uses the scale parameter to calculate the correct dimensions, to avoid doing this transformation twice, we _divide_ by this value to calculate the correct dimensions for the request parameters. This is needed so that we can maintain API compatibility with older client versions.

Without this workaround, the scale will be applied twice, resulting in larger images than necessary (and longer download times, more data usage, and decreased performance).

This PR also fetches the screenshots for the theme picker using the dimensions from the "recommended" sizes (rather than "category" sizes). Other changes this PR relies on are on the API side (`D83271-code`).

### Testing steps

Pre-conditions:

* <s>Device should be connected to a WPCOM sandbox</s> (The back-end changes have been deployed).
* <s>The patch should be applied to the sandbox from `D83271-code`</s>

1. Navigate to the site creation flow
2. Choose an Intent (e.g. "Food")
3. Expect the recommended for you screenshots to be larger
4. Expect the screenshots to be loaded with proper display resolution (i.e. not pixelated)
   * It can be helpful to inspect traffic (e.g. in Android Studio network inspector) to ensure that properly sized screenshots are requested.
   * Inspecting traffic can also help confirm that the returned screenshots are the correct size (limited to 1200 width).

#### Page picker

The page picker should not be affected by these changes (though backend changes in the API may result in changes to the fetched screenshot URLs)

## Regression Notes
1. Potential unintended areas of impact
Page picker

6. What I did to test those areas of impact (or what existing automated tests I relied on)
Manual tests

7. What automated tests I added (or what prevented me from doing so)
Establishing the required testing apparatus for this would be out of scope for this task.

PR submission checklist:
